packages/statedb/src/interfaces.ts

```diff
@@ -29,6 +29,13 @@ import { ISignal } from '@lumino/signaling';
  * method. Defaults to `string`;
  */
 export interface IDataConnector<T, U = T, V = string, W = string> {
+  /**
+   * Response type of the data connector as a string.
+   * Used as a discriminant if a property has multiple connectors
+   * with different response types.
+   */
+  responseType?: string;
```

Comment:
My preference I think is this to not go in the main IDataConnector type, but keeping it localized to the completer code. I added a comment up there to suggest how it might be done?